apollo-backend/src/main/java/io/logz/apollo/services/SlaveService.java

```diff
@@ -59,13 +61,18 @@ public SlaveService(ApolloConfiguration apolloConfiguration, SlaveDao slaveDao,
         keepaliveExecutorService = Executors.newSingleThreadScheduledExecutor(new BasicThreadFactory.Builder()
                                             .namingPattern("slave-keepalive-pinger")
                                             .build());
+
+        cleanupUnusedSlavesExecutorService = Executors.newSingleThreadScheduledExecutor(new BasicThreadFactory.Builder()
+                                                      .namingPattern("cleanup-unused-slaves-service")
+                                                      .build());
     }
 
     @PostConstruct
     public void start() {
         logger.info("Starting slave.. slaveId - {}, environmentIds - {}", slaveId, apolloConfiguration.getSlave().getSlaveCsvEnvironments());
 
-        cleanupUnusedSlaves();
+        cleanupUnusedSlavesExecutorService.scheduleWithFixedDelay(this::cleanupUnusedSlaves,
```

Comment:
1. you must have forgotten to shutdown `cleanupUnusedSlavesExecutorService`
this class has `stop()` function. check the line 88

2. why do you creating executor service here and not inside if statement? (line 77)

Reply:
1. correct I forgot..
2. because I want to cleanup unused slaves no matter what the status of the specific slave.. maybe it isn't the place to do this.. maybe we should do it when starting apollo

Reply:
ok. cool
can we restrict cleanup ExecutorService to run only on master?
I think scheduling this task on each instance is redundant.

Reply:
how about putting this section under if(!isSlave) statement?

Reply:
sounds good